src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -361,15 +509,38 @@ Result Swapchain::Present(
         return PresentHeadless(imageIndex, waitSemaphoreCount, ppWaitSemaphores);
     }
 
-    return PresentInternal(imageIndex, waitSemaphoreCount, ppWaitSemaphores);
+    if (IsIndirect()) {
+        RecordPreamble(imageIndex);
+    }
+    uint32_t                      nextWaitSemaphoreCount = waitSemaphoreCount;
+    const grfx::Semaphore* const* ppNextWaitSemaphores   = ppWaitSemaphores;
+    if (mIsRecording[imageIndex]) {
```

Comment:
could you add some comments to explain what you want to achieve here? if `IsIndirect()` is false, `mIsRecording[imageIndex]` will always be false?

Reply:
Not necessarily. It's intended to keep track if there's any post processing (e.g. imgui).